File: py2many/add_method_1.jl
struct Foo
_value::
end

function __init__{T0}(self::Foo, value::T0)
self._value = value
end

function __add__{T0}(self::Foo, other::T0)::Foo
return Foo(self._value + other._value)
end

function __str__(self::Foo)::String
return "*"*self._value
end

File: py2many/add_method_1.cpp
class Foo {
public:
  auto _value;

  template <typename T0> void __init__(T0 value) { auto this->_value = value; }

  template <typename T0> Foo __add__(T0 other) {
    return Foo((this->_value) + (other._value));
  }

  inline std::string __str__() { return std::string{"*"} * (this->_value); }
};
